From c1ce964f3e9312100a60f03c1e1fdd601e1911f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C4=90o=C3=A0n=20Tr=E1=BA=A7n=20C=C3=B4ng=20Danh?=
 <congdanhqx@gmail.com>
Date: Tue, 28 Feb 2023 18:45:15 +0100
Subject: [PATCH] test: Fix a race condition in merge.test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Just like commit 4dfe7f9, (test: Fix a race condition, 2023-01-20),
this fix a test race when stdout and stderr in any order.

Upstream-Status: Backport [https://git.savannah.nongnu.org/cgit/quilt.git/commit/?id=c1ce964f3e9312100a60f03c1e1fdd601e1911f2]
Signed-off-by: Đoàn Trần Công Danh <congdanhqx@gmail.com>
Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 test/merge.test | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/test/merge.test b/test/merge.test
index c64b33d..2e67d4f 100644
--- a/test/merge.test
+++ b/test/merge.test
@@ -39,8 +39,9 @@ Test the patch merging functionality of `quilt diff'.
 	> Applying patch %{P}c.diff
 	> Now at patch %{P}c.diff
 
-	$ quilt diff -P b.diff | grep -v "^\\(---\\|+++\\)"
+	$ quilt diff -P b.diff >/dev/null
 	> Warning: more recent patches modify files in patch %{P}b.diff
+	$ quilt diff -P b.diff 2>/dev/null | grep -v "^\\(---\\|+++\\)"
 	>~ Index: [^/]+/abc\.txt
 	> ===================================================================
 	> @@ -1,3 +1,3 @@
@@ -49,8 +50,9 @@ Test the patch merging functionality of `quilt diff'.
 	> +b+
 	>  c
 
-	$ quilt diff --combine a.diff -P b.diff | grep -v "^\\(---\\|+++\\)"
+	$ quilt diff --combine a.diff -P b.diff >/dev/null
 	> Warning: more recent patches modify files in patch %{P}b.diff
+	$ quilt diff --combine a.diff -P b.diff 2>/dev/null | grep -v "^\\(---\\|+++\\)"
 	>~ Index: [^/]+/abc\.txt
 	> ===================================================================
 	> @@ -1,3 +1,3 @@
-- 
2.40.0

